### PR TITLE
MAT-7155 wrong type for EncounterPerformed:diagnoses

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR Template

Jira Ticket: [MAT-7155](https://jira.cms.gov/browse/MAT-7155)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [ ] Required tests are included.
* [x] No extemporaneous files are included (i.e. Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java [hunk 1]
```diff
@@ -40,7 +41,7 @@ public class EncounterPerformed extends DataElement {
   private Interval relevantPeriod;
   private Code dischargeDisposition;
   private List<Code> facilityLocations;
-  private List<Code> diagnoses;
+  private List<Component> diagnoses;
```
Comment (adongare) [inline src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: I think this was always incorrect. Correct attribute would be DiagnosisComponent which is different from Component.
```suggestion
  private List<DiagnosisComponent> diagnoses;
```
image

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: @adongare Yes you are right, this is the data: 
image
I used a more generic class and it seems working fine: 
image
Let me know if you think we should create a new class for DiagnosisComponent, thanks Ashok!

Comment (adongare) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: yeah, that is incorrect. diagnosis should have these fields.
image

Comment (adongare) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: and if in case we did not implement support for this in UI, we'll have to enter the tickets

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: @adongare Ok thanks so much for the feedback, I'll create a new class for DiagnosisComponent.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/EncounterPerformed.java]: @adongare I created and used DiagnosisComponent for EncounterPerformed. Thanks!

File: src/main/java/gov/cms/madie/models/cqm/datacriteria/DiagnosisComponent.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DiagnosisComponent {
+  private String qdmVersion = "5.6";
+  private String _type = "QDM::DiagnosisComponent";
+  private String _id;
+  private Code code;
+  private Code presentOnAdmissionIndicator;
+  private int rank;
```
Comment (adongare) [inline src/main/java/gov/cms/madie/models/cqm/datacriteria/DiagnosisComponent.java]: looks good. have a question though- `int` type would be initialized to 0 by default which is a kind of valid rank. Should we use the wrapper type `private Integer rank` instead so that null would be allowed?

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/models/cqm/datacriteria/DiagnosisComponent.java]: @adongare Good point! I just changed it, thanks!

**Review by adongare — APPROVED** [review]
